LanguageSpec.md

```diff
@@ -1151,227 +1151,162 @@ Formally, for a function "f(x1: t1, x2: t2, ..., xn: tn)" and input arguments 
-
-Gets all values stored in the specified dictionary and for a simple list it returns all elements.
-
-#####IndexOf:int(list: var[]..[], element: var[]..[])`
-
-Returns the index of the member in the list.
-
-#####Insert:var[]..[](list: var[]..[], element: var, index: int)`
-
-Inserts an element into a list at specified index.
-
-#####Insert:var[]..[](list: var[]..[], element: var[]..[], index: int)`
-
-Inserts an element into a list at specified index.
-
-#####IsRectangular: bool(list: var[]..[])`
-
-Checks if each of rows in multidimensional  list has the same number of elements.
-
-#####ImportFromCSV:double[][](filePath: string )`
-
-Imports data from a text file containing comma separated values into two-dimensional list.
-
-#####ImportFromCSV:double[][](filePath: string, transpose:bool)`
-
-Imports data from a text file containing comma separated values into two-dimensional list and 
-
-also transpose the output list if specified.
-
-#####IsHomogeneous: bool(list: var[]..[])`
-
-Checks if all the elements in the specified list are of the same type.
-
-#####IsUniformDepth:bool(list: var[]..[])`
-
-Checks if the list has a uniform depth.
-
-#####Map:double(rangeMin: double, rangeMax: double, inputValue: double)`
-
-Maps a value into an input range.
-
-#####MapTo:double(rangeMin: double, rangeMax: double, inputValue: double, targetRangeMin: double, targetRangeMax:double)`
-
-Maps a value from one range to another range.
-
-#####NormalizeDepth:var[]..[](list: var[]..[])`
-
-Returns a list with uniform depth as specified by the input depth.
-
-#####NormalizeDepth:var[]..[](list: var[]..[], rank: var)`
-
-Return multidimensional list according to the rank given.
-
-#####Print(msg: var)`
-
-Print msg to the console. 
-
-#####Rank(list: var[]..[])`
-
-Counts the maximal rank of the specified list. 
-
-#####Remove:var(list: var[]..[], index: int)`
-
-Removes element at the specified index of the list.
-
-#####RemoveDuplicates:var[]..[](list: var[]..[])`
+```
+a = 1;
+b = NameOf( a ); // "number"
+```
 
-Removes duplicate elements in the specified list.
+```
+a = {};
+b = NameOf( a ); // "dictionary"
+```
 
-#####RemoveNulls:var[]..[](list: var[]..[])`
+### Dictionaries
 
-Removes null elements from the specified list.
+#### Modification
 
-#####RemoveIfNot:var[]..[](list: var[]..[], type:string)`
+##### `Append(dictionary : var[]..[], value: var[]..[]) : var[]..[] `
 
-Removes the members of the list which are not members of the specified type.
+`Append` creates a new `dictionary` with a new element inserted at the end. If the `dictionary` is not array-like, returns an `error`.
 
-#####RemoveKey:bool(list:var[]..[], key: var)`
+Examples:
 
-Returns true if the specified key is removed from the specified list; otherwise returns false.
+```
+a = {1, 2, 3};
+b = Append(a, 4); // {1,2,3,4}
+```
 
-#####Reorder:var[](list: var[], indice:var[])`
+##### `Set(dictionary : var[]..[], key : var, value : var) : var[]..[]`
```

Comment:
I don't think so. Such an operation would require first extracting the nested list and then setting the value.